### ISSUE TITLE
Test Program stuck while grab

### ISSUE THREAD

**Issue by xiuxiu** [cedricve/raspicam#6]
Hi,

I built raspicam successfully but failed to ran the test program.
The program stuck like the following:

pi@raspberrypi:~/workspace/raspicam/build/utils $ ./raspicam_test
Usage (-help for help)
Connecting to camera
Connected to camera =0000000006384f06 bufs=3686400
Capturing....

I have no idea about what is happening, please help.

**Comment by FranciscMoldovan** [comment]
Hi!
Does your camera work with the provided `raspistill` program from `rasbpian`?
please try out `raspistill -o image.jpg`. Sometimes these cameras fry due to their sensibility to ESD.

**Comment by BNB1** [comment]
Hi xiuixiu and others,

actually, I have they exact same problem. Could you solve this problem? I would be really interested in a solution.
In this german raspi [forum](http://www.forum-raspberrypi.de/Thread-noir-kamera-modul-c), a guy said that in the CMakeLists.txt which is in the src directory the following highlighted header is missing:

SET(public_hdrs_base raspicamtypes.h raspicam.h **raspicam_still.h**)

But I still have the same problem, when I add "raspicam_still.h", so it must be something else. Any ideas?
Or is there another way of taking full resolution pictures in c++?
Thank you all in advance!

**Comment by birgersp** [comment]
@BNB1 the header includes are unrelated, the compiler would complain if there were any headers missing.

Did you remember to enable the camera in [this menu](https://www.raspberrypi.org/documentation/configuration/raspi-config.md)?

**Comment by BNB1** [comment]
@birgersp: thank you for your reply.
Yes I definitely enabled the camera in the menu. And of course, my raspi is up to date. I still could not solve this problem.

**Comment by descampsa** [comment]
I had the same problem, but it appeared only with the experimental opengl option enabled. Try to disable it if it is enabled.
After a complete update of my raspbian, the problem seems to have disappeard, not sure which package fixed it.

**Comment by BNB1** [comment]
Finally, I had the time to try your suggestion. Thank you for that. Unfortunately I'm not sure if I understand you right. Which opengl option do you mean? The one in the advanced settings of the raspi-config? Then this was disabled already. In addition to that, I have not install any opengl package manually (if that is possible).

**Comment by descampsa** [comment]
Yes, i was talking about the opengl option in the advanced settings of the raspi-config. 
If it was already disabled, i have no idea what your problem is, unfortunately.
Maybe try to run raspicam with strace to see what system call fails, if any.

**Comment by BNB1** [comment]
Mmh. I just wanted to enable and then disable the opengl setting (I know, kind of a desperate act...) but since then, my Raspi does not boot anymore... I mean, I see the colored screen, then the screen, where I can press shift but afterward nothing happens. Oh, oh. Hope that I can recover my data on the SD card somehow...

**Comment by descampsa** [comment]
Sorry about that, the opengl driver are still experimental, so i guess that this can happen. I also had to disable it because it made my raspberry quite unstable.

Have you tried to access it by ssh, maybe it is just the display that freezes, but the system boots correctly?

**Comment by BNB1** [comment]
No, I have never done this before. But I already started reading me into this subject to try this as a next step.

**Comment by descampsa** [comment]
Another option, maybe easier, would be to mount your sd card on another computer and disable the opengl driver manually by editing the appropriate config files. 

You have to comment the `dtoverlay=vc4-kms-v3d` line in /boot/config.txt, delete the file `/etc/xdg/autostart/xcompmgr.desktop`, and rename `/usr/share/X11/xorg.conf.d/99-fbturbo.~` to `/usr/share/X11/xorg.conf.d/99-fbturbo.conf`. 
This what raspi-config does to disable the driver (from https://github.com/RPi-Distro/raspi-config/blob/master/raspi-config#L1170).

**Comment by BNB1** [comment]
Oh my goodness, you saved my evening!! I could not find the files to delete/rename, but apperently, editing the config file is sufficient!! Thank you very much!
